README.md
jskom
=====

jskom is a web based LysKOM client written in Javascript. It comes
with a Python web app for configuring and serving the Javascript
files.

jskom uses httpkom for communication with the LysKOM server.

jskom source code is available at: https://github.com/osks/jskom


Dependencies
------------

httpkom: https://github.com/osks/httpkom

For required Python packages, see requirements.txt. Install them with:

    $ pip install -r requirements.txt

For running the tests from the Makefile in a console, you need
PhantomJS. You can still run the tests in a regular browser if you
want. (If you use Homebrew on OS X, you can install it with 'brew
install phantomjs'.)


For information: The jskom tests uses the Mocha framework
(https://mochajs.org/) and mocha-phantomjs
(http://metaskills.net/mocha-phantomjs/). Both are included with
jskom. For mocha-phantomjs to work, mocha.js had to be patched to use
a real diff library:

    require.register("browser/diff.js", function(module, exports, require){
      module.exports = JsDiff; // this row was added
    });


Authors
-------

Oskar Skoog <oskar@osd.se>


Copyright and license
---------------------

Copyright (c) 2012-2014 Oskar Skoog. jskom is provided under the MIT
license. See the included LICENSE.txt file for specifics.


CHANGELOG.md
# Changelog for jskom

## Unreleased

### Added

- Mode for showing a text as ROT13.
- Showing in the footer which server the current session is on.
- Grouping sessions by server in the sessions menu.

### Changed

- Update dependencies to newer Quart and Hypercorn, and thus also
  newer httpkom and newer Flask. And pin Werkzeug <3.
- Use DispatcherMiddleware from Hypercorn instead of our own, which no
  longer worked after updating Hypercorn.
- Python 3.7 is minimum. 3.6 is not supported (wasn't supported before
  either, the information in setup.py was incorrect).


## 0.26 (2022-09-12)

### Added

- UI for changing the presentation of the current person.
- UI for changing the password of the current person.

### Fixed

- Update httpkom and pylyskom versions to handle reading texts by
  anonymous users and comments to texts by anonymous users.


## 0.25 (2021-01-21)

### Fixed

- More work towards avoiding bad browser caching. Index page should
  not be cached now and API requests should also use the static
  version in URLs.

### Changed

- The initial fetch of membership now fetches a few (20) in the first
  request, and then fetches more (200) per request. This is to make
  the web page feel faster when logging in, because it's quite slow to
  fetch memberships after logging in (as non of the uconfs are cached
  by httpkom).


## 0.24 (2021-01-20)

### Fixed

- Update httpkom version to fix serialization issues


## 0.23 (2021-01-20)

### Fixed

- Fixed pylyskom version in requirements.txt so the Docker image can build.


## 0.22 (2021-01-20)

### Changed

- Updates to work with next versions of httpkom/pylyskom (conf_name -> name).


## 0.21 (2020-07-03)

### Added

- Command line arguments for enabling debug logging (now off by default).

### Fixed

- Incorrect use of async methods.

### Changed

- Updated required httpkom version to 0.17


## 0.20 (2020-07-03)

### Added

- Command line arguments for which host and port to listen on.

### Fixed

- Dockerfile improvements.


## 0.19 (2020-07-01)

### Fixed

- Exclude webassets cache files in python dist.


## 0.18 (2020-07-01)

### Changed

- Updated required httpkom version to 0.16.
- Dockerfile improvements.
- Minor changes.


## 0.17 (2020-02-07)

### Fixed

- Fix version handling. It wasn't possible to install httpkom due to
  the way we tried to single source the package version.

### Changed

- Run with python3 -m jskom
- Converted from using Flask to Quart (and Hypercorn), to be able to
  include the httpkom API in the same server process. Uses the asyncio
  version of httpkom / pylyskom.

### Added

- Jskom now includes the httpkom API in the same webserver and it's
  used by default. This is simplify the setup and to avoid having to
  run httpkom separately (separate cert, process management, etc). It
  is still possible to continue using jskom with a separate httpkom
  server (using the HTTPKOM_SERVER variable).


## 0.16 (2020-01-24)

- New "Go to text" page (under the Texts menu).
- Now uses Referrer Policy "origin-when-cross-origin".
- Updated versions for Flask and Werkzeug dependencies.
- Published on PyPI.


## 0.15 (2020-01-24)

(Same as 0.16 but was discarded because a faulty dist was uploaded to
PyPI.)


## 0.14 (2016-05-29)

### Added

- Setting version on what has been working okay for several years now.
- Added changelog.
- mx-date and mx-author handling.
- Support for showing link to conference FAQ text.

### Fixed

- Fixed various caching issues.
- Fixed bug #45.

### Changed

- Probably some other things I have forgotten about.


## 0.13 (2013-02-11)

## 0.12 (2013-01-08)

## 0.11 (2012-12-18)

## 0.10 (2012-12-02)

## 0.9 (2012-10-21)


komauxitems.py
# AUTOGENERATED by make_komauxitems from aux-items.py

AI_CONTENT_TYPE = 1 # (text)
AI_FAST_REPLY = 2 # (text)
AI_CROSS_REFERENCE = 3 # (text, conference, letterbox)
AI_NO_COMMENTS = 4 # (text)
AI_PERSONAL_COMMENT = 5 # (text)
AI_REQUEST_CONFIRMATION = 6 # (text)
AI_READ_CONFIRM = 7 # (text)
AI_REDIRECT = 8 # (conference, letterbox)
AI_X_FACE = 9 # (conference, letterbox, server)
AI_ALTERNATE_NAME = 10 # (text, conference, letterbox)
AI_PGP_SIGNATURE = 11 # (text)
AI_PGP_PUBLIC_KEY = 12 # (letterbox)
AI_E_MAIL_ADDRESS = 13 # (conference, letterbox, server)
AI_FAQ_TEXT = 14 # (conference, letterbox, server)
AI_CREATING_SOFTWARE = 15 # (text)
AI_MX_AUTHOR = 16 # (text)
AI_MX_FROM = 17 # (text)
AI_MX_REPLY_TO = 18 # (text)
AI_MX_TO = 19 # (text)
AI_MX_CC = 20 # (text)
AI_MX_DATE = 21 # (text)
AI_MX_MESSAGE_ID = 22 # (text)
AI_MX_IN_REPLY_TO = 23 # (text)
AI_MX_MISC = 24 # (text)
AI_MX_ALLOW_FILTER = 25 # (conference, letterbox)
AI_MX_REJECT_FORWARD = 26 # (conference, letterbox)
AI_NOTIFY_COMMENTS = 27 # (letterbox)
AI_FAQ_FOR_CONF = 28 # (text)
AI_RECOMMENDED_CONF = 29 # (server)
AI_ALLOWED_CONTENT_TYPE = 30 # (conference, letterbox, server)
AI_CANONICAL_NAME = 31 # (server)
AI_MX_LIST_NAME = 32 # (conference)
AI_SEND_COMMENTS_TO = 33 # (letterbox)
AI_WORLD_READABLE = 34 # (text)
AI_MX_REFUSE_IMPORT = 35 # (conference, letterbox)
AI_ELISP_CLIENT_READ_FAQ = 10000 # (letterbox)
AI_ELISP_CLIENT_REJECTED_RECOMMENDATION = 10001 # (letterbox)
AI_MX_MIME_BELONGS_TO = 10100 # (text)
AI_MX_MIME_PART_IN = 10101 # (text)
AI_MX_MIME_MISC = 10102 # (text)
AI_MX_ENVELOPE_SENDER = 10103 # (text)
AI_MX_MIME_FILE_NAME = 10104 # (text)
AI_MX_ALLOW_ENVELOPE_SENDER_REGEXP = 10105 # (conference, letterbox)

aux_item_number_to_name = {
  AI_CONTENT_TYPE: "content-type",
  AI_FAST_REPLY: "fast-reply",
  AI_CROSS_REFERENCE: "cross-reference",
  AI_NO_COMMENTS: "no-comments",
  AI_PERSONAL_COMMENT: "personal-comment",
  AI_REQUEST_CONFIRMATION: "request-confirmation",
  AI_READ_CONFIRM: "read-confirm",
  AI_REDIRECT: "redirect",
  AI_X_FACE: "x-face",
  AI_ALTERNATE_NAME: "alternate-name",
  AI_PGP_SIGNATURE: "pgp-signature",
  AI_PGP_PUBLIC_KEY: "pgp-public-key",
  AI_E_MAIL_ADDRESS: "e-mail-address",
  AI_FAQ_TEXT: "faq-text",
  AI_CREATING_SOFTWARE: "creating-software",
  AI_MX_AUTHOR: "mx-author",
  AI_MX_FROM: "mx-from",
  AI_MX_REPLY_TO: "mx-reply-to",
  AI_MX_TO: "mx-to",
  AI_MX_CC: "mx-cc",
  AI_MX_DATE: "mx-date",
  AI_MX_MESSAGE_ID: "mx-message-id",
  AI_MX_IN_REPLY_TO: "mx-in-reply-to",
  AI_MX_MISC: "mx-misc",
  AI_MX_ALLOW_FILTER: "mx-allow-filter",
  AI_MX_REJECT_FORWARD: "mx-reject-forward",
  AI_NOTIFY_COMMENTS: "notify-comments",
  AI_FAQ_FOR_CONF: "faq-for-conf",
  AI_RECOMMENDED_CONF: "recommended-conf",
  AI_ALLOWED_CONTENT_TYPE: "allowed-content-type",
  AI_CANONICAL_NAME: "canonical-name",
  AI_MX_LIST_NAME: "mx-list-name",
  AI_SEND_COMMENTS_TO: "send-comments-to",
  AI_WORLD_READABLE: "world-readable",
  AI_MX_REFUSE_IMPORT: "mx-refuse-import",
  AI_ELISP_CLIENT_READ_FAQ: "elisp-client-read-faq",
  AI_ELISP_CLIENT_REJECTED_RECOMMENDATION: "elisp-client-rejected-recommendation",
  AI_MX_MIME_BELONGS_TO: "mx-mime-belongs-to",
  AI_MX_MIME_PART_IN: "mx-mime-part-in",
  AI_MX_MIME_MISC: "mx-mime-misc",
  AI_MX_ENVELOPE_SENDER: "mx-envelope-sender",
  AI_MX_MIME_FILE_NAME: "mx-mime-file-name",
  AI_MX_ALLOW_ENVELOPE_SENDER_REGEXP: "mx-allow-envelope-sender-regexp",
}
